backend/server.js

```diff
@@ -241,12 +244,18 @@ sbp('okTurtles.data/set', PUBSUB_INSTANCE, createServer(hapi.listener, {
     // partition
     const recoveredState = Object.create(null)
     recoveredState.contracts = Object.create(null)
+    const channels = sbp('okTurtles.data/get', PUBSUB_INSTANCE).channels
     await Promise.all(savedStateIndex.split('\x00').map(async (contractID) => {
       const cpSerialized = await sbp('chelonia/db/get', `_private_cheloniaState_${contractID}`)
-      if (!cpSerialized) return
+      if (!cpSerialized) {
+        console.warn('[chelonia] Contract in index but missing state', contractID)
+        return
+      }
       const cp = JSON.parse(cpSerialized)
```

Comment:
What does `cp` mean in `cpSerialized` and `cp`? From the above comment I would guess `contractPartition`, but I'm not sure, and without the comment I would have no clue.

Is the concept of contract partition explained somewhere? `git grep partition` only yields this one occurence in that comment.

Reply:
Yes, it means contract partition and no, it's not explained elsewhere, but it refers to how the state is partitioned by contract (hence, contract partition). The reason to do this is performance.